test/Base_Tests/src/Data/Decimal_Spec.enso

```diff
@@ -0,0 +1,478 @@
+            (Decimal.new 3.4 . add (Decimal.new 20.05) (Math_Context.new 3 Rounding_Mode.bankers)) . should_equal (Decimal.new 23.4)
+            (Decimal.new -3.4 . add (Decimal.new -20.05) (Math_Context.new 3)) . should_equal (Decimal.new -23.5)
+            (Decimal.new -3.4 . add (Decimal.new -20.05) (Math_Context.new 3 Rounding_Mode.bankers)) . should_equal (Decimal.new -23.4)
+            (Decimal.new -867.625 . subtract (Decimal.new -33.05) (Math_Context.new 5)) . should_equal (Decimal.new -834.58)
+            (Decimal.new -867.625 . subtract (Decimal.new -33.05) (Math_Context.new 5 Rounding_Mode.bankers)) . should_equal (Decimal.new -834.58)
+            (Decimal.new 49.31 . multiply (Decimal.new 5) (Math_Context.new 4)) . should_equal (Decimal.new 246.6)
+            (Decimal.new 49.31 . multiply (Decimal.new 5) (Math_Context.new 4 Rounding_Mode.bankers)) . should_equal (Decimal.new 246.6)
+            (Decimal.new 49.29 . multiply (Decimal.new 5) (Math_Context.new 4)) . should_equal (Decimal.new 246.5)
+            (Decimal.new 49.29 . multiply (Decimal.new 5) (Math_Context.new 4 Rounding_Mode.bankers)) . should_equal (Decimal.new 246.4)
+            (Decimal.new 247.75 . divide (Decimal.new -5) (Math_Context.new 3)) . should_equal (Decimal.new -49.6)
+            (Decimal.new 247.75 . divide (Decimal.new -5) (Math_Context.new 3 Rounding_Mode.bankers)) . should_equal (Decimal.new -49.6)
+
+        group_builder.specify "should allow mixed arithmetic" <|
+            (Decimal.new 1 + 2) . should_equal 3
+            (Decimal.new 1 - 2) . should_equal -1
+            (Decimal.new 3 * 4) . should_equal 12
+            (Decimal.new 3 / 4) . should_equal 0.75
+
+            (1 + Decimal.new 2) . should_equal 3
+            (1 - Decimal.new 2) . should_equal -1
+            (3 * Decimal.new 4) . should_equal 12
+            (3 / Decimal.new 4) . should_equal 0.75
+
+            (Decimal.new 1 + 2.0) . should_equal 3
+            (Decimal.new 1 - 2.0) . should_equal -1
+            (Decimal.new 3 * 4.0) . should_equal 12
+            (Decimal.new 3 / 4.0) . should_equal 0.75
+
+            (1 + Decimal.new 2.0) . should_equal 3
+            (1 - Decimal.new 2.0) . should_equal -1
+            (3 * Decimal.new 4.0) . should_equal 12
+            (3 / Decimal.new 4.0) . should_equal 0.75
+
+        group_builder.specify "should get an aritmetic error when a result is a nonterminating decimal expansion" <|
+            (Decimal.new 1 / Decimal.new 3) . should_fail_with Arithmetic_Error
+
+        group_builder.specify "should negate values correctly" <|
+            Decimal.new 5 . negate . should_equal -5
+            Decimal.new -5 . negate . should_equal 5
+            Decimal.new 5.3 . negate . should_equal -5.3
+            Decimal.new -5.3 . negate . should_equal 5.3
+            Decimal.new 0 . negate . should_equal 0
+            Decimal.new -0 . negate . should_equal 0
```

Comment:
What would we need to add to be able to support
```
d = Decimal.new 4
x = -d
x.should_equal -4
```

Reply:
This is already supported; added tests for this format.

Reply:
Cool!